localstack/services/ec2/resource_providers/aws_ec2_keypair.py

```diff
@@ -0,0 +1,128 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+
+
+class EC2KeyPairProperties(TypedDict):
+    KeyName: Optional[str]
+    KeyFingerprint: Optional[str]
+    KeyFormat: Optional[str]
+    KeyPairId: Optional[str]
+    KeyType: Optional[str]
+    PublicKeyMaterial: Optional[str]
+    Tags: Optional[list[Tag]]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class EC2KeyPairProvider(ResourceProvider[EC2KeyPairProperties]):
+    TYPE = "AWS::EC2::KeyPair"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[EC2KeyPairProperties],
+    ) -> ProgressEvent[EC2KeyPairProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/KeyName
+
+        Required properties:
+          - KeyName
+
+        Create-only properties:
+          - /properties/KeyName
+          - /properties/KeyType
+          - /properties/KeyFormat
+          - /properties/PublicKeyMaterial
+          - /properties/Tags
+
+        Read-only properties:
+          - /properties/KeyPairId
+          - /properties/KeyFingerprint
+
+        IAM permissions required:
+          - ec2:CreateKeyPair
+          - ec2:ImportKeyPair
+          - ec2:CreateTags
+          - ssm:PutParameter
```

Comment:
kinda curious about the put parameter here.

From the AWS docs:

> When you create a new key pair, the private key is saved to AWS Systems Manager Parameter Store, using a parameter with the following name: /ec2/keypair/{key_pair_id}. For more information about retrieving private key, and the required permissions, see [Create a key pair using AWS CloudFormation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/create-key-pairs.html#create-key-pair-cloudformation) in the Amazon EC2 User Guide.

Not super urgent but seems this would be missing in terms of parity with the resource on AWS. Basically we need to create a parameter here and delete it again in the delete call.

Reply:
Thanks I just added that behavior to resource provider